### ISSUE TITLE
[Filter field] Async behaviour is buggy when current filter is deleted while loading.

### ISSUE THREAD

**Issue by danielkaneider** [dynatrace-oss/barista#1436]
# Bug Report

## Steps to Reproduce

- open the StackBlitz example below
- working case:
-- select 'AUT' via keyboard
-- wait 2 secs
-- select 'Linz'
- not working case:
-- select 'AUT' via keyboard
-- immediately press the back button
-- once the data is loaded, you lose the key. Moreover, if you clear all entries, it will show you the cities instead of the countries

## Expected Behavior

To have 'AUT: Linz' in the filter field

## Current Behavior

It just lists the cities

## Possible Solution
Allow to set the root object into the datasource, eg. 
```
  private ASYNC_DATA = {
    autocomplete: [
      {
        name: 'AUT (async)',
        autocomplete: [{ name: 'Linz' }, { name: 'Vienna' }, { name: 'Graz' }]
      },
      {
        name: 'USA',
        autocomplete: [
          { name: 'San Francisco' },
          { name: 'Los Angeles' },
          { name: 'New York' },
        ],
      },
    ],
  };
```
https://github.com/dynatrace-oss/barista/issues/1438

## Context (Environment)

https://stackblitz.com/github/dynatrace-oss/barista-examples/tree/master/examples/DtExampleFilterFieldAsync/

## Attachments
![image](https://user-images.githubusercontent.com/1572080/89381590-b7f97080-d6f9-11ea-9e1d-f7537d8054d1.png)

**Comment by tomheller** [comment]
Hi @danielkaneider!
This is a nice find, thank you for reporting this. Would you be willing to contribute a fix for this problem as well?

**Comment by thomaspink** [comment]
Thank you for reporting. Let me summarise our conclusion after a short call:
The basic issue is, that the consumer has to cancel the ongoing request if the consumer deletes the current async node. If not, the filter field switches back to the root data when the user hits backspace and when the async data in then provided to the data source, the data is seen as new root data because the async node has been deleted by the user.

In short: A pending request has to be canceled when the user removed the corresponding inital async node.

In addition to that, we need to discuss if we wanna[ remove the possibility of a user-interaction while the filter-field is waiting for data](#1464).
